addons/mail/models/mail_guest.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import pytz
+import uuid
+from odoo.tools import consteq
+from odoo import fields, models, _
+from odoo.addons.base.models.res_partner import _tz_get
+
+class MailGuest(models.Model):
+    _name = 'mail.guest'
+    _description = "Guest"
+
+    access_token = fields.Char(
+        copy=False,
+        default=lambda self: str(uuid.uuid4()),
+        groups='base.group_system',
+        readonly=True,
+        required=True,
+    )
+    channel_ids = fields.Many2many(comodel_name='mail.channel')
+    country_id = fields.Many2one(string="Country", comodel_name='res.country')
+    lang_id = fields.Many2one(string="Language", comodel_name='res.lang')
```

Comment:
why do you need a many2one? it seems you are only using it to retrieve the code so can just have a selection field `lang` as we do from time to time

Reply:
copy/paste from website visitor is the reason

We'll look into changing it if you think there is a better way.